db/migrate/20161128200400_update_quotes.rb

```diff
@@ -0,0 +1,11 @@
+class UpdateQuotes < ActiveRecord::Migration
+  def change
+    change_column_null :quotes, :anime_id, false
+    change_column_null :quotes, :user_id, false
+    change_column_null :quotes, :content, false
+
+    add_foreign_key :quotes, :users
+    add_foreign_key :quotes, :anime
+    add_foreign_key :quotes, :characters
```

Comment:
Well, considering `user_id` and `anime_id` already exist, I guess the easiest would be:
```ruby
# Delete all entries with incomplete dataset
Quote.where(anime_id: nil).destroy_all
Quote.where(user_id: nil).destroy_all

# Merge anime_id data over to media_id
rename_column :quotes, :anime_id, :media_id

# Add type for polymorphic relationship
add_column :quotes, :media_type, :string, default: 'Anime', null: false

# Makes sure no more invalid data can be added
change_column_null :quotes, :user_id, false

# Add reference to characters table
add__reference :quotes, :character, foreign_key: true
```

Just a write-up from the top of my head, maybe @NuckChorris can confirm/correct this real quick.